executor/builder.go

```diff
@@ -1236,15 +1236,79 @@ func (b *executorBuilder) buildUpdate(v *plan.Update) Executor {
 		b.err = errors.Trace(b.err)
 		return nil
 	}
+	columns2Handle := buildColumns2Handle(v.Schema(), tblID2table)
 	updateExec := &UpdateExec{
-		baseExecutor: newBaseExecutor(b.ctx, nil, v.ExplainID(), selExec),
-		SelectExec:   selExec,
-		OrderedList:  v.OrderedList,
-		tblID2table:  tblID2table,
+		baseExecutor:   newBaseExecutor(b.ctx, nil, v.ExplainID(), selExec),
+		SelectExec:     selExec,
+		OrderedList:    v.OrderedList,
+		tblID2table:    tblID2table,
+		columns2Handle: columns2Handle,
 	}
 	return updateExec
 }
 
+// Columns2HandleEntry represents an mapper from column index to handle index.
```

Comment:
how about:
```go
// Columns2HandleEntry maps a consecutive columns to a handle column.
type Columns2HandleEntry struct {
    start, end    int32 // Represent the ordinal range [start, end) of the consecutive columns.
    handleOrdinal int32 // Represents the ordinal of the handle column.
}
```

And it seems that `start` is always equal to `handleIdx`, do we need to store `handleIdx`?

Reply:
`handleOrdinal` may equal to `start` or `end` o.o?

if table without the primary key, the handle will be `end`
if table with the primary key, the handle will be  primary definition index, e.g. `create table t5(v int, k int, primary key(k))`, will be start=1, end=3, handle=2

Reply:
If so, I think to use getTableOffset to decide the begin offset of a table is wrong:
```go
199 func getTableOffset(schema *expression.Schema, handleCol *expression.Column) int {
200     for i, col := range schema.Columns {
201         if col.DBName.L == handleCol.DBName.L && col.TblName.L == handleCol.TblName.L {
202             return i
203         }
204     }
205     panic("Couldn't get column information when do update/delete")
206 }
```

Reply:
from this three logic:

https://github.com/lysu/tidb/blob/44a2ad7f0f65e5c5ea3dab0f20a34fe5a357e58a/plan/logical_plan_builder.go#L1817

https://github.com/lysu/tidb/blob/44a2ad7f0f65e5c5ea3dab0f20a34fe5a357e58a/plan/point_get_plan.go#L260

https://github.com/lysu/tidb/blob/44a2ad7f0f65e5c5ea3dab0f20a34fe5a357e58a/plan/find_best_task.go#L312

handle in `[start, end]`.....

for `getTableOffset`..it will works if schema.Columns is order by columns group that grouped by table...  e.g.  `|t1.x, t1.y| t2.z, t2,u|`....

this logic is same to https://github.com/lysu/tidb/blob/e5922f4ff012e748bb51bd7e854fe02a9813fd93/executor/update.go#L45 update#exec